### ISSUE TITLE
docs: make GHA default, docker optional

### ISSUE THREAD

**Issue by casperdcl** [iterative/cml#482]
See e.g. https://partner.github.com/integration-resources/2020/11/24/pattern-setting-up-a-cli-on-github-s-hosted-runners.html

Would make it super easy to set up:

```yaml
jobs:
  run:
    runs-on: [ubuntu-latest]
    steps:
      - uses: actions/checkout@v2
      - uses: iterative/cml@v1
      - name: 'Train my model'
        ...
        cml-send-comment report.md
```

EDIT:

```yaml
- uses: actions/setup-node@v1
- uses: iterative/setup-cml@v1
```
works well but needs better documentation. Any objections to making it more prominent than `container: docker://`? Would fix e.g. #360

**Comment by 0x2b3bfa0** [comment]
👋 Hi, @casperdcl! Is [`uses: iterative/setup-cml@v1`](https://github.com/iterative/setup-cml) what you are looking for?

**Comment by casperdcl** [comment]
Aha yes; seems the docs really push for using docker instead.

I would discourage that thoroughly. There's no way people with complex build envs (most users?) would prefer a custom docker image over a native action.

**Comment by 0x2b3bfa0** [comment]
Agreed! If I recall correctly, @DavidGOrtega recommended Docker since the beginning because of isolation, but maybe we can explicitly offer both alternatives.

**Comment by casperdcl** [comment]
Docker's a great options, but is far from a canonical use case of GHA. Even if docker is used, it probably refers to `action.yml` using docker rather than a workflow doing `container: docker://...`

**Comment by 0x2b3bfa0** [comment]
Definitely, this may be worth a few documentation updates. What do you think, @DavidGOrtega?

> Docker's a great option, but is far from a canonical use case of GHA.

Agreed! At least, when used for the same purposes as the usual [`setup-$TOOL`](https://partner.github.com/integration-resources/2020/11/24/pattern-setting-up-a-cli-on-github-s-hosted-runners.html) actions you mentioned.

> Even if docker is used, it probably refers to `action.yml` using docker rather than a workflow doing `container: docker://...`

Our GitHub Action is JavaScript-based and doesn't use Docker at all; it wouldn't add any value to the current use cases.

**Comment by casperdcl** [comment]
> Our GitHub Action is JavaScript-based and doesn't use Docker at all; it wouldn't add any value to the current use cases.

Yes I know, I'm not suggesting changing `action.yml`. I just mean that `container: docker://` is fairly unexpected even for GHA users who know about Docker.

**Comment by casperdcl** [comment]
I think the current README is a bit outdated and repetitive, hiding important info.

Missing and/or hidden things:

- https://github.com/iterative/setup-cml
  - `[uses: actions/setup-node@v1, uses: iterative/setup-cml@v1]`
- https://dvc.org/blog/dvc-2-0-release#new-method-to-provision-cloud-compute-in-new-cml-release

I can open a PR for more concrete discussion

/CC @shcheklein

**Comment by 0x2b3bfa0** [comment]
> I just mean that `container: docker://` is fairly unexpected even for GHA users who know about Docker.

👍  It's practical for some use cases, but not specially common among users, and probably not a sensible default for many complex workflows.

> _I think the current README is a bit outdated and repetitive, hiding important info._

Version 0.3.0 introduced some drastic changes, so I wouldn't be surprised if certain sections leave room for improvement. 😅 

> _I can open a PR for more concrete discussion_

That would be awesome!

**Comment by DavidGOrtega** [comment]
@casperdcl @0x2b3bfa0 
actions/setup-node@v1 might not be needed at all since its de facto in GHA.

I encourage using docker of course but not our image in GHA but the user's. Thats why we changed CML a bit.
Our cml.dev examples pushes the user to use the action instead our container but seems that we forgot this here?

I wont think that this fix #360 unfortunately since we have to support Gitlab and that job (better tagging) has to be done anyway.

**Comment by casperdcl** [comment]
Well it's needed if you want to pin the version or on often in `self-hosted` but I'm sure you knew that. Will certainly point it out in the docs...
